workflows/index-generation/generate_lineage_csvs.py

```diff
@@ -141,6 +144,30 @@ def _equals(previous_row: Dict[str, str], row: Dict[str, str]):
     return True
 
 
+def _find_highest_taxonomy_changed(previous_row: Dict[str, str], row: Dict[str, str]):
+    """
+    Finds the highest rank taxonomy level at which a taxon was reclassified. e.g. if species A was reassigned
+    to a different genus and family, it would return the new family name instead of the genus name.
+    """
+    fieldnames_to_search = [
```

Comment:
Could you use `_fieldnames` here?

Reply:
The reason I didn't was that I wanted to traverse them in a particular order (level name, level id, is phage, tax id, tax name) from highest to lowest rank. This is so that we find the highest level at which the taxonomy changed